button label text

joshbousel opened the pull request.

noahdklein committed:
button label text

joshbousel commented on main.js:
```diff
@@ -49,11 +49,17 @@ $(function(){
 		} else {
 			formHTML += '<input type="'+$input.attr('type')+'" class="input-text input-full" name="'+id+'" id="'+id+'">';
 		}
+		var buttonLabel = $input.attr('data-button-label');
```
@noahdklein buttonLabel shouldn't be an attribute of the input, it should be on the $emailContainer object.
To simplify this I'd condense into one line added after line 5 (to keep the data attrs pulled from the container together):

`var buttonLabel = ($emailContainer.attr('data-button-label') != 'undefined') ? $emailContainer.attr('data-button-label') : 'Submit';`

noahdklein committed:
no message

noahdklein committed:
no message

noahdklein committed:
no message

noahdklein reviewed (commented):
I removed the quotes around the button label in this line which was throwing an error 
formHTML += '<a href="#" class="btn-large btn-large--submit fill-callout inline-block">'+buttonLabel+'</a>';

on line 5 I changed teh != to == and it seems to now accept the undefined state,  however I'm not sure how to target the data-label-button in the html since it's not an input or a button.  wouldn't we have to target the class to get it to recognize the change?

joshbousel commented on main.js:
```diff
@@ -3,6 +3,7 @@ var $ = require('jquery');
 $(function(){
 	var $emailContainer = $('.convio-survey');
 	var surveyID = $emailContainer.attr('data-surveyid');
+	var buttonLabel = ($emailContainer.attr('data-button-label') == 'undefined') ? $emailContainer.attr('data-button-label') : 'Submit';
```
@noahdklein The logic here is now wrong. It's says that if the attribute data-button-label is undefined, then use the data-button-label attribute as the button label, and if it isn't undefined, use "Submit." You want the opposite, which was why I had it as !=.

It's looking for the data-button-label label on the object with the class "convio-survey," so that's where you need to place the attribute in the html. IE:

`<div class="convio-survey" data-surveyid="14288" data-button-label="Get My Certificate" style="display: none;">`

noahdklein committed:
changed operator back to !=
Removed quotes around undefined which was reading and inserting it as a string.

joshbousel commented:
@noahdklein Has this now been tested?

noahdklein commented:
I tested the default and the data-label button.  Both are displaying and passing data correctly to the survey.
https://www.wcs.org/elephant-origami-certificate?_preview=true

joshbousel commented:
Great! You can merge it in.
